logstash-core/lib/logstash/api/commands/base.rb

```diff
@@ -19,6 +19,43 @@ def uptime
         def started_at
           (LogStash::Agent::STARTED_AT.to_f * 1000.0).to_i
         end
+
+        # Return a hash including the values of the keys given at the path given
+        # 
+        # Example Usage:
+        # extract_metrics(
+        #   [:jvm, :process],
+        #   :open_file_descriptors,
+        #   [:cpu, [:total_in_millis, :percent]]
+        # )
+        # 
+        # Returns:
+        # # From the jvm.process metrics namespace
+        # {
+        #   :open_file_descriptors => 123
+        #   :cpu => { :total_in_millis => 456, :percent => 789 }
+        # }
+        def extract_metrics(path, *keys)
```

Comment:
Yes. @ph proposed moving this into the collector. I agree with him on this. I'll wait till #5366 is addressed before I attempt that move.